apps/api/src/campaign/campaign.service.ts

```diff
@@ -603,29 +603,28 @@ export class CampaignService {
     return await this.prisma.$transaction(async (tx) => {
       let donationId
       // Find donation by extPaymentIntentId
-      const existingDonation = await this.findExistingDonation(tx, paymentData)
+      const existingPayment = await this.findExistingDonation(tx, paymentData)
 
       //if missing create the donation with the incoming status
-      if (!existingDonation) {
+      if (!existingPayment) {
         const newDonation = await this.createIncomingDonation(
           tx,
           paymentData,
           newDonationStatus,
           campaign,
         )
-        donationId = newDonation.id
+        donationId = newDonation.donations[0].id
```

Comment:
is it always just 1 donation? can this be 0 or more than 1?

Reply:
At this point it will always be 1 donation.